base/set.jl

```diff
@@ -191,6 +191,62 @@ function unique(f::Callable, C)
 end
 
 """
+    unique!(c)
+Remove duplicate items and return the modified collection, as determined by
+[`isequal`](@ref).
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A);
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+```
+"""
+function unique!(c)
+    if issorted(c) || issorted(reverse(c))
+        idxs = eachindex(c)
+        # When we loop through c, we only need to remember one state and one
+        # item in the collection.
+        currentstate = start(idxs)
+        currentitem = c[currentstate]
```

Comment:
If you limit this to `AbstractVector` then you can use `for i in indices(v,1)` to iterate from the first index to the last. I would argue that `AbstractVector` is the only built-in data structure that it makes sense to define this for. For ordered collections the "keys" are integers, in which case the collection is a kind of `AbstractVector`. The only other built-in general collection types are `Dict` and `Set` and `unique!` doesn't make sense for those.